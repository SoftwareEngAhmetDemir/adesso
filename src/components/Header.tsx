import { NavLink } from 'react-router-dom';
interface HeaderTypes {
  title: string;
}
function Header({ title }: HeaderTypes) {
  return (
    <header id="header">
      <nav className="navbar navbar-expand-lg  bg-white d-flex justify-content-between pt-3">
        <div className="container-fluid custom-navbar">
          <h1 className="title">{title}</h1>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <div className="d-flex" role="search">
              <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                <li className="nav-item">
                  <NavLink className="nav-link" to="/">
                    Home
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink className="nav-link" to="/units">
                    Units
                  </NavLink>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </nav>
    </header>
  );
}

export default Header;
