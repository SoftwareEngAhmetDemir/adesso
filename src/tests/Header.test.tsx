import { shallow } from 'enzyme';
import Header from '../components/Header';

test('should test Header component', () => {
 const wrapper = shallow(<Header title={''} />);
 expect(wrapper).toMatchSnapshot();
});