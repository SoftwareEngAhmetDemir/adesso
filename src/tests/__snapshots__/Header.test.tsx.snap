// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Header component 1`] = `
<header
  id="header"
>
  <nav
    className="navbar navbar-expand-lg  bg-white d-flex justify-content-between pt-3"
  >
    <div
      className="container-fluid custom-navbar"
    >
      <h1
        className="title"
      />
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <div
          className="d-flex"
          role="search"
        >
          <ul
            className="navbar-nav me-auto mb-2 mb-lg-0"
          >
            <li
              className="nav-item"
            >
              <NavLink
                className="nav-link"
                to="/"
              >
                Home
              </NavLink>
            </li>
            <li
              className="nav-item"
            >
              <NavLink
                className="nav-link"
                to="/units"
              >
                Units
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
